### ISSUE TITLE
Generated records shouldn't have a @ConstructorProperties annotation on their POJO constructor

### ISSUE THREAD

**Issue by agentgt** [jOOQ/jOOQ#13319]
### Expected behavior

Generated records should have the correct number of `@ConstructorProperties` for POJO to record constructor.

### Actual behavior

When a record class is generated and POJO creation is turned on as well as generate constructors the constructor that takes the POJO has the same `@ConstructorProperties` as the field based constructor.

```java
    @ConstructorProperties({ "id", "authorId", "coAuthorId", "detailsId", "title", "publishedIn", "languageId", "contentText", "contentPdf", "recVersion", "recTimestamp" })
    public BookRecord(Integer id, Integer authorId, Integer coAuthorId, Integer detailsId, String title, Integer publishedIn, Integer languageId, String contentText, byte[] contentPdf, Integer recVersion, LocalDateTime recTimestamp) {
//snip
    }

    @ConstructorProperties({ "id", "authorId", "coAuthorId", "detailsId", "title", "publishedIn", "languageId", "contentText", "contentPdf", "recVersion", "recTimestamp" })
    public BookRecord(com.snaphop.test.db.h2.tables.pojos.Book value) {
//snip
    }
```

**A couple of our annotation processors including MapStruct and some custom ones get angry about the mismatch of parameters. Like will not compile.**

### Steps to reproduce the problem

```xml
 <generate>
   <pojos>true</pojos>                      
   <constructorPropertiesAnnotationOnPojos>true</constructorPropertiesAnnotationOnPojos>
   <constructorPropertiesAnnotationOnRecords>true</constructorPropertiesAnnotationOnRecords>
 </generate>
```

Examine generated code.

### Versions

- jOOQ: 3.16.5
- Java: 17

**Comment by agentgt** [comment]
It looks like you just need some logic checking if `pojoArgument` here:

https://github.com/jOOQ/jOOQ/blob/a8b4fa62316aca9873d998a61f2d736c32b85a87/jOOQ-codegen/src/main/java/org/jooq/codegen/JavaGenerator.java#L2108

Hopefully that helps. Let me know if you want do a whole PR (I didn't because I don't have the cycles at the moment).

**Comment by lukaseder** [comment]
Thanks a lot for your report. I can reproduce that.

> Let me know if you want do a whole PR (I didn't because I don't have the cycles at the moment).

Usually, a clear report like yours is optimal. PRs often lack details. Given that you don't have access to the test suite, there's also a certain regression risk. See also: https://github.com/jOOQ/jOOQ/blob/main/CONTRIBUTING.md

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.6 (https://github.com/jOOQ/jOOQ/issues/13320), and 3.15.10 (https://github.com/jOOQ/jOOQ/issues/13321).

Thanks again for your report!
